### ISSUE TITLE
Getting a warning for the ea-hub that's not in the edsc-hub

### ISSUE THREAD

**Issue by lwasser** [earthlab/hub-ops#226]
I think this might be a chart version issue...  but i'll open this in case we can update the ea-hub to remove that warning for future builds. 
```
NFO:root:Executing helm upgrade --install --namespace ea-hub ea-hub hub-charts/ea-hub --force --wait --timeout 600 --cleanup-on-fail -f secrets/ea-hub.yaml --set-string jupyterhub.singleuser.image.tag=b9239ef

2020/06/03 21:52:25 Warning: Merging destination map for chart 'jupyterhub'. The destination item 'lifecycleHooks' is a table and ignoring the source 'lifecycleHooks' as it has a non-table value of: <nil>

2020/06/03 21:52:25 Warning: Merging destination map for chart 'jupyterhub'. The destination item 'lifecycleHooks' is a table and ignoring the source 'lifecycleHooks' as it has a non-table value of: <nil>
```

**Comment by lwasser** [comment]
there is one error in the logs
```
(base) CIRES-EL-LM-020:hub-ops leahwasser$ kubectl -n ea-hub get events --sort-by='{.lastTimestamp}'
LAST SEEN   TYPE      REASON                   OBJECT                                 MESSAGE
28m         Normal    Killing                  pod/jupyter-lherwehe                   Stopping container notebook
9m33s       Normal    Scheduled                pod/jupyter-stephlshep                 Successfully assigned ea-hub/jupyter-stephlshep to gke-jhub-user-pool-10b9aebf-gk5m
9m28s       Normal    SuccessfulAttachVolume   pod/jupyter-stephlshep                 AttachVolume.Attach succeeded for volume "pvc-5013befd-cb4f-11e9-b882-42010a80017d"
9m11s       Normal    Pulled                   pod/jupyter-stephlshep                 Container image "earthlabhubops/ea-k8s-user-ea-hub:b9239ef" already present on machine
9m11s       Normal    Started                  pod/jupyter-stephlshep                 Started container block-cloud-metadata
9m11s       Normal    Created                  pod/jupyter-stephlshep                 Created container block-cloud-metadata
9m11s       Normal    Pulled                   pod/jupyter-stephlshep                 Container image "jupyterhub/k8s-network-tools:0.8.2" already present on machine
9m10s       Normal    Started                  pod/jupyter-stephlshep                 Started container notebook
9m10s       Normal    Created                  pod/jupyter-stephlshep                 Created container notebook
3m9s        Normal    SuccessfulCreate         daemonset/hook-image-puller            Created pod: hook-image-puller-s5w4f
3m9s        Normal    Scheduled                pod/hook-image-puller-97rlx            Successfully assigned ea-hub/hook-image-puller-97rlx to gke-jhub-user-pool-10b9aebf-gk5m
3m9s        Normal    Scheduled                pod/hook-image-puller-s5w4f            Successfully assigned ea-hub/hook-image-puller-s5w4f to gke-jhub-core-pool-6318399a-c678
3m9s        Normal    SuccessfulCreate         daemonset/hook-image-puller            Created pod: hook-image-puller-97rlx
3m8s        Normal    Created                  pod/hook-image-puller-97rlx            Created container image-pull-metadata-block
3m8s        Normal    Started                  pod/hook-image-puller-97rlx            Started container image-pull-singleuser
3m8s        Normal    Started                  pod/hook-image-puller-97rlx            Started container image-pull-metadata-block
3m8s        Normal    Created                  pod/hook-image-puller-97rlx            Created container image-pull-singleuser
3m8s        Normal    Pulled                   pod/hook-image-puller-s5w4f            Container image "jupyterhub/k8s-network-tools:0.8.2" already present on machine
3m8s        Normal    Pulled                   pod/hook-image-puller-97rlx            Container image "jupyterhub/k8s-network-tools:0.8.2" already present on machine
3m8s        Normal    Pulled                   pod/hook-image-puller-97rlx            Container image "earthlabhubops/ea-k8s-user-ea-hub:b9239ef" already present on machine
3m7s        Normal    Pulled                   pod/hook-image-puller-97rlx            Container image "gcr.io/google_containers/pause:3.0" already present on machine
3m7s        Normal    Created                  pod/hook-image-puller-97rlx            Created container pause
3m7s        Normal    Started                  pod/hook-image-puller-s5w4f            Started container image-pull-metadata-block
3m7s        Normal    Created                  pod/hook-image-puller-s5w4f            Created container image-pull-metadata-block
3m7s        Normal    Pulled                   pod/hook-image-puller-s5w4f            Container image "gcr.io/google_containers/pause:3.0" already present on machine
3m6s        Normal    Started                  pod/hook-image-puller-s5w4f            Started container pause
3m6s        Normal    Created                  pod/hook-image-puller-s5w4f            Created container pause
3m6s        Normal    Started                  pod/hook-image-puller-97rlx            Started container pause
3m6s        Normal    Killing                  pod/hook-image-puller-97rlx            Stopping container pause
3m5s        Normal    SandboxChanged           pod/hook-image-puller-s5w4f            Pod sandbox changed, it will be killed and re-created.
3m5s        Normal    Pulled                   pod/hook-image-puller-s5w4f            Container image "earthlabhubops/ea-k8s-user-ea-hub:b9239ef" already present on machine
3m5s        Normal    Created                  pod/hook-image-puller-s5w4f            Created container image-pull-singleuser
3m5s        Normal    SandboxChanged           pod/hook-image-puller-97rlx            Pod sandbox changed, it will be killed and re-created.
3m4s        Normal    Scheduled                pod/hook-image-puller-zcrkg            Successfully assigned ea-hub/hook-image-puller-zcrkg to gke-jhub-user-pool-10b9aebf-gk5m
3m4s        Warning   FailedCreatePodSandBox   pod/hook-image-puller-97rlx            Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hook-image-puller-97rlx": Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"process_linux.go:413: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: time=\\\\\\\"2020-06-03T21:52:33Z\\\\\\\" level=fatal msg=\\\\\\\"no such file or directory\\\\\\\"\\\\n\\\"\"": unknown
3m4s        Normal    SuccessfulCreate         daemonset/hook-image-puller            Created pod: hook-image-puller-zlr7v
3m4s        Normal    Started                  pod/hook-image-puller-s5w4f            Started container image-pull-singleuser
3m4s        Normal    SuccessfulCreate         daemonset/hook-image-puller            Created pod: hook-image-puller-zcrkg
3m4s        Normal    Scheduled                pod/hook-image-puller-zlr7v            Successfully assigned ea-hub/hook-image-puller-zlr7v to gke-jhub-core-pool-6318399a-c678
3m3s        Normal    Pulled                   pod/hook-image-puller-zlr7v            Container image "earthlabhubops/ea-k8s-user-ea-hub:b9239ef" already present on machine
3m3s        Normal    Created                  pod/hook-image-puller-zlr7v            Created container image-pull-singleuser
3m3s        Normal    Created                  pod/hook-image-puller-zcrkg            Created container image-pull-singleuser
3m3s        Normal    Pulled                   pod/hook-image-puller-zcrkg            Container image "earthlabhubops/ea-k8s-user-ea-hub:b9239ef" already present on machine
3m2s        Normal    Started                  pod/hook-image-puller-zlr7v            Started container image-pull-metadata-block
3m2s        Normal    Created                  pod/hook-image-puller-zlr7v            Created container image-pull-metadata-block
3m2s        Normal    Started                  pod/hook-image-puller-zcrkg            Started container image-pull-singleuser
3m2s        Normal    Pulled                   pod/hook-image-puller-zlr7v            Container image "jupyterhub/k8s-network-tools:0.8.2" already present on machine
3m2s        Normal    Started                  pod/hook-image-puller-zlr7v            Started container image-pull-singleuser
3m1s        Normal    Created                  pod/hook-image-puller-zcrkg            Created container image-pull-metadata-block
3m1s        Normal    Created                  pod/hook-image-puller-zlr7v            Created container pause
3m1s        Normal    Pulled                   pod/hook-image-puller-zlr7v            Container image "gcr.io/google_containers/pause:3.0" already present on machine
3m1s        Normal    Pulled                   pod/hook-image-puller-zcrkg            Container image "jupyterhub/k8s-network-tools:0.8.2" already present on machine
3m1s        Normal    Scheduled                pod/hook-image-awaiter-mt7l7           Successfully assigned ea-hub/hook-image-awaiter-mt7l7 to gke-jhub-user-pool-10b9aebf-gk5m
3m1s        Normal    SuccessfulCreate         job/hook-image-awaiter                 Created pod: hook-image-awaiter-mt7l7
3m1s        Normal    Started                  pod/hook-image-puller-zcrkg            Started container image-pull-metadata-block
3m          Normal    Started                  pod/hook-image-puller-zcrkg            Started container pause
3m          Normal    Created                  pod/hook-image-puller-zcrkg            Created container pause
3m          Normal    Pulled                   pod/hook-image-puller-zcrkg            Container image "gcr.io/google_containers/pause:3.0" already present on machine
3m          Normal    Started                  pod/hook-image-puller-zlr7v            Started container pause
2m59s       Normal    Started                  pod/hook-image-awaiter-mt7l7           Started container hook-image-awaiter
2m59s       Normal    Pulled                   pod/hook-image-awaiter-mt7l7           Container image "jupyterhub/k8s-image-awaiter:0.8.2" already present on machine
2m59s       Normal    Created                  pod/hook-image-awaiter-mt7l7           Created container hook-image-awaiter
2m58s       Normal    Killing                  pod/hook-image-puller-zcrkg            Stopping container pause
2m58s       Normal    Killing                  pod/hook-image-puller-zlr7v            Stopping container pause
2m45s       Normal    Scheduled                pod/hook-image-awaiter-m2b68           Successfully assigned ea-hub/hook-image-awaiter-m2b68 to gke-jhub-user-pool-10b9aebf-gk5m
2m45s       Normal    SuccessfulCreate         job/hook-image-awaiter                 Created pod: hook-image-awaiter-m2b68
2m44s       Normal    Created                  pod/hook-image-awaiter-m2b68           Created container hook-image-awaiter
2m44s       Normal    Pulled                   pod/hook-image-awaiter-m2b68           Container image "jupyterhub/k8s-image-awaiter:0.8.2" already present on machine
2m44s       Normal    Started                  pod/hook-image-awaiter-m2b68           Started container hook-image-awaiter
2m43s       Normal    SuccessfulCreate         job/hook-image-awaiter                 Created pod: hook-image-awaiter-bznsh
2m43s       Normal    Scheduled                pod/hook-image-awaiter-bznsh           Successfully assigned ea-hub/hook-image-awaiter-bznsh to gke-jhub-user-pool-10b9aebf-gk5m
2m42s       Normal    Created                  pod/hook-image-awaiter-bznsh           Created container hook-image-awaiter
2m42s       Normal    Pulled                   pod/hook-image-awaiter-bznsh           Container image "jupyterhub/k8s-image-awaiter:0.8.2" already present on machine
2m41s       Normal    Started                  pod/hook-image-awaiter-bznsh           Started container hook-image-awaiter
2m36s       Normal    NoPods                   poddisruptionbudget/user-scheduler     No matching pods found
2m36s       Normal    NoPods                   poddisruptionbudget/user-placeholder   No matching pods found
2m33s       Normal    Scheduled                pod/hook-image-awaiter-rml24           Successfully assigned ea-hub/hook-image-awaiter-rml24 to gke-jhub-user-pool-10b9aebf-gk5m
2m33s       Normal    SuccessfulCreate         job/hook-image-awaiter                 Created pod: hook-image-awaiter-rml24
2m32s       Normal    Started                  pod/hook-image-awaiter-rml24           Started container hook-image-awaiter
2m32s       Normal    Pulled                   pod/hook-image-awaiter-rml24           Container image "jupyterhub/k8s-image-awaiter:0.8.2" already present on machine
2m32s       Normal    Created                  pod/hook-image-awaiter-rml24           Created container hook-image-awaiter
113s        Normal    SuccessfulCreate         job/hook-image-awaiter                 Created pod: hook-image-awaiter-kj9jj
113s        Normal    Scheduled                pod/hook-image-awaiter-kj9jj           Successfully assigned ea-hub/hook-image-awaiter-kj9jj to gke-jhub-user-pool-10b9aebf-gk5m
112s        Normal    Created                  pod/hook-image-awaiter-kj9jj           Created container hook-image-awaiter
112s        Normal    Pulled                   pod/hook-image-awaiter-kj9jj           Container image "jupyterhub/k8s-image-awaiter:0.8.2" already present on machine
112s        Normal    Started                  pod/hook-image-awaiter-kj9jj           Started container hook-image-awaiter
33s         Normal    Scheduled                pod/hook-image-awaiter-kmfwt           Successfully assigned ea-hub/hook-image-awaiter-kmfwt to gke-jhub-user-pool-10b9aebf-gk5m
33s         Normal    SuccessfulCreate         job/hook-image-awaiter                 Created pod: hook-image-awaiter-kmfwt
32s         Normal    Pulled                   pod/hook-image-awaiter-kmfwt           Container image "jupyterhub/k8s-image-awaiter:0.8.2" already present on machine
32s         Normal    Created                  pod/hook-image-awaiter-kmfwt           Created container hook-image-awaiter
32s         Normal    Started                  pod/hook-image-awaiter-kmfwt           Started container hook-image-awaiter
```

**Comment by lwasser** [comment]
`3m4s        Warning   FailedCreatePodSandBox   pod/hook-image-puller-97rlx            Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hook-image-puller-97rlx": Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"process_linux.go:413: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: time=\\\\\\\"2020-06-03T21:52:33Z\\\\\\\" level=fatal msg=\\\\\\\"no such file or directory\\\\\\\"\\\\n\\\"\"": unknown` it would be good to understand what these logs generally mean.

**Comment by lwasser** [comment]
the chart version is one thing that seems to be different. what i don't know is if it's dangerous to attempt to update it while we have an active class?? maybe we should avoid any updates now?

**Comment by lwasser** [comment]
it also failed to deploy
```
UPGRADE FAILED

Error: timed out waiting for the condition

Error: UPGRADE FAILED: timed out waiting for the condition

Do not need to rebuild image, using earthlabhubops/ea-k8s-user-ea-hub:b9239ef

Using earthlabhubops/ea-k8s-user-ea-hub:b9239ef as user image for ea-hub

Traceback (most recent call last):

  File "./deploy.py", line 291, in <module>

    main()

  File "./deploy.py", line 287, in main

    deploy(args.chartname)

  File "./deploy.py", line 228, in deploy

    helm(*install_args)

  File "./deploy.py", line 13, in helm

    return subprocess.check_call(['helm'] + list(args), **kwargs)

  File "/opt/python/3.7.1/lib/python3.7/subprocess.py", line 341, in check_call

    raise CalledProcessError(retcode, cmd)

subprocess.CalledProcessError: Command '['helm', 'upgrade', '--install', '--namespace', 'ea-hub', 'ea-hub', 'hub-charts/ea-hub', '--force', '--wait', '--timeout', '600', '--cleanup-on-fail', '-f', 'secrets/ea-hub.yaml', '--set-string', 'jupyterhub.singleuser.image.tag=b9239ef']' returned non-zero exit status 1.

The command "# Stage 3c, Step 2: Deploy the earth-analytics-hub

      python ./deploy.py --build --push --deploy ea-hub

      " failed and exited with 1 during .
```
oddly i wasn't updating the ea-hub so i'm not sure what that is all about

**Comment by lwasser** [comment]
i think i fixed this via #229 !!! closing as it's deploying nicely now :)
